src/db_iterator.rs

```diff
@@ -529,6 +529,9 @@ impl Iterator for DBWALIterator {
     type Item = (u64, WriteBatch);
 
     fn next(&mut self) -> Option<(u64, WriteBatch)> {
+        if !self.valid() {
```

Comment:
Adding this check as the test will crash without this check.

Also, according to [include/rocksdb/iterator.h](https://github.com/yhchiang-sol/rocksdb/blob/main/include/rocksdb/iterator.h#L65-L68), valid is required before calling next.